foreman/client.py

```diff
@@ -368,6 +364,25 @@ def parse_resource_definition(resource_name, resource_dct):
     return new_dict, foreign_methods
 
 
+resource_pattern = re.compile(r'^/api(/v[12])?/(?P<resource>\w+).*')
+
+
+def parse_resource_from_url(url):
```

Comment:
Can you move this as staticmethod to the above class?

Also the resource_pattern as a class property